File: go-macaroon/basic_macaroon_signature.go
// script arguments: location key id

package main

import (
  "os"
  "fmt"

  "gopkg.in/macaroon.v1"
)


func main() {
    args := os.Args[1:]

    key := []byte(args[1])

    m, err := macaroon.New(key, args[2], args[0])

    if err == nil {
      fmt.Printf("%x\n", m.Signature())
    }
}

File: implementations/rust-macaroons/basic_macaroon_signature.rs
// script arguments: location key id

extern crate macaroons;

use std::env;
use std::string::String;

use macaroons::token::{Token, Tag};

fn to_hex_string(bytes: Vec<u8>) -> String {
  let strs: Vec<String> = bytes.iter()
                               .map(|b| format!("{:x}", b))
                               .collect();
  strs.connect("")
}

fn main() {
  let args: Vec<_> = env::args().collect();
  let location = args[1].clone().into_bytes();
  let key = args[2].clone().into_bytes();
  let id = args[3].clone().into_bytes();


  let token = Token::new(key, id, location);
  let Tag(sig) = token.tag;

  let signature = to_hex_string(sig.to_vec());

  println!("{}", &signature);
}
